faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -919,6 +919,26 @@ class ClientSpec
       identifyR.to[Boolean].get shouldBe true
       }
 
+  it should "test CurrentToken with internal token" in {
+    val createR = client.query(Create(Collection("spells"), Obj("credentials" -> Obj("password" -> "abcdefg")))).futureValue
```

Comment:
Just a quick suggestion: I know the style within the tests is a bit messy at the moment, but I wanted to share a quick _layout_ that works for me. :)

If possible, I usually try to divide the test cases with comments in three sections: _Setup_, _Run_ and _Verify_. This way, it's instantly clear what is happening to others looking at the code (and especially for future me! 😄).

For this case, it would look like something like below:

```scala
// Setup
val createR = client.query(Create(Collection("spells"), Obj("credentials" -> Obj("password" -> "abcdefg")))).futureValue
val loginR = client.query(Login(createR(RefField), Obj("password" -> "abcdefg"))).futureValue
val secret = loginR(SecretField).get
val tokenRef = loginR(RefField).get

// Run
val currentToken = client.sessionWith(secret)(_.query(CurrentToken())).futureValue

// Verify
currentToken.to[RefV].get shouldBe tokenRef
```

Reply:
Well, doing some googling, I remembered this is called the "AAA Pattern": [The fundamentals of unit testing: Arrange, Act, Assert](https://defragdev.com/blog/2014/08/07/the-fundamentals-of-unit-testing-arrange-act-assert.html).